languagetool-standalone/src/test/java/org/languagetool/ProtoResultMatchCacheTest.java

```diff
@@ -0,0 +1,360 @@
+      RuleMatch deserializedResultMatch = CacheUtils.deserializeResultMatch(cachedResultMatch, ruleMatch.getSentence());
+
+      assertEquals(ruleMatch.getRule().getId(), deserializedResultMatch.getRule().getId());
+      assertEquals(ruleMatch.getRule().getFullId(), deserializedResultMatch.getRule().getFullId());
+      assertEquals(ruleMatch.getRule().getSubId(), deserializedResultMatch.getRule().getSubId());
+      assertEquals(ruleMatch.getRule().getDescription(), deserializedResultMatch.getRule().getDescription());
+      assertEquals(ruleMatch.getRule().estimateContextForSureMatch(), deserializedResultMatch.getRule().estimateContextForSureMatch());
+      assertEquals(ruleMatch.getRule().getSourceFile(), deserializedResultMatch.getRule().getSourceFile());
+      assertEquals(ruleMatch.getRule().getLocQualityIssueType(), deserializedResultMatch.getRule().getLocQualityIssueType());
+      assertEquals(ruleMatch.getRule().isDefaultTempOff(), deserializedResultMatch.getRule().isDefaultTempOff());
+      assertEquals(ruleMatch.getRule().getCategory().getName(), deserializedResultMatch.getRule().getCategory().getName());
+      assertEquals(ruleMatch.getRule().isPremium(), deserializedResultMatch.getRule().isPremium());
+      assertEquals(ruleMatch.getRule().getTags(), deserializedResultMatch.getRule().getTags());
+
+      assertEquals(ruleMatch.getMessage(), deserializedResultMatch.getMessage());
+      assertEquals(ruleMatch.getShortMessage(), deserializedResultMatch.getShortMessage());
+      assertEquals(ruleMatch.getPatternFromPos(), deserializedResultMatch.getPatternFromPos());
+      assertEquals(ruleMatch.getPatternToPos(), deserializedResultMatch.getPatternToPos());
+      assertEquals(ruleMatch.getFromPos(), deserializedResultMatch.getFromPos());
+      assertEquals(ruleMatch.getToPos(), deserializedResultMatch.getToPos());
+      assertEquals(ruleMatch.getFromPosSentence(), deserializedResultMatch.getFromPosSentence());
+      assertEquals(ruleMatch.getToPosSentence(), deserializedResultMatch.getToPosSentence());
+      assertEquals(ruleMatch.getSuggestedReplacementObjects(), deserializedResultMatch.getSuggestedReplacementObjects());
+      assertEquals(ruleMatch.getUrl() != null ? ruleMatch.getUrl() : ruleMatch.getRule().getUrl(), deserializedResultMatch.getUrl());
+      assertEquals(ruleMatch.getType(), deserializedResultMatch.getType());
+      assertEquals(ruleMatch.isAutoCorrect(), deserializedResultMatch.isAutoCorrect());
+      assertEquals(ruleMatch.getNewLanguageMatches(), deserializedResultMatch.getNewLanguageMatches());
+      assertEquals(ruleMatch.getSpecificRuleId(), deserializedResultMatch.getSpecificRuleId());
+      assertEquals(ruleMatch.getOriginalErrorStr(), deserializedResultMatch.getOriginalErrorStr());
+      assertEquals(ruleMatch.getSentence(), deserializedResultMatch.getSentence());
+
+
+      assertEquals(ruleMatch, deserializedResultMatch);
```

Comment:
Can you test out [reflectionEquals](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/EqualsBuilder.html#reflectionEquals(java.lang.Object,java.lang.Object,boolean)) for this as well to make sure we've covered everything?